e2e/protractor/suites/actions/copy-move/copy.test.ts

```diff
@@ -246,42 +246,72 @@ describe('Copy content', () => {
       undoCopyFolderWithExistingName(folderExisting, '', destinationPF));
   });
 
-  async function copyFile(fileName: string, location: string = '', destination: string, doBefore?: () => void) {
-    if (doBefore) {
-      await doBefore();
+  async function baseCopy(
+    itemName: string | string[],
+    location: string,
+    destination: string,
+    undo: boolean = false,
+    expectedMsg: string = 'Copied 1 item'
+  ) {
+    if (itemName instanceof Array) {
+      await dataTable.selectMultipleItems(itemName, location);
+    } else {
+      await dataTable.selectItem(itemName, location);
     }
-    await dataTable.selectItem(fileName, location);
     await toolbar.clickMoreActionsCopy();
     await copyDialog.selectLocation('Personal Files');
     await copyDialog.selectDestination(destination);
     await BrowserActions.click(copyDialog.copyButton);
     const msg = await page.getSnackBarMessage();
-    expect(msg).toContain('Copied 1 item');
+    expect(msg).toContain(expectedMsg);
     const action = await page.getSnackBarAction();
     expect(action).toContain('Undo');
+    if (undo) {
+      await page.clickSnackBarAction();
+      await page.clickPersonalFilesAndWait();
+    } else {
+      await copyDialog.waitForDialogToClose();
+    }
+  }
 
-    await copyDialog.waitForDialogToClose();
+  async function baseCopyDoBefore(
+    itemName: string | string[],
+    location: string,
+    destination: string,
+    doBefore?: () => void,
+    undo?: boolean,
+    expectedMsg?: string
+  ) {
+    if (doBefore) {
+      doBefore();
+    }
+    await baseCopy(itemName, location, destination, undo, expectedMsg);
+  }
+
+  async function baseCopyDoBeforeAsync(
+    itemName: string | string[],
+    location: string,
+    destination: string,
+    doBefore?: () => Promise<void>,
+    undo?: boolean,
+    expectedMsg?: string
+  ) {
+    if (doBefore) {
+      await doBefore();
+    }
+    await baseCopy(itemName, location, destination, undo, expectedMsg);
+  }
+
+  async function copyFile(fileName: string, location: string = '', destination: string, doBefore?: () => void) {
```

Comment:
can we remove type from location as default value is applied?

Reply:
done